pkg/manager/internal.go

```diff
@@ -357,6 +359,18 @@ func (cm *controllerManager) GetWebhookServer() *webhook.Server {
 	return cm.webhookServer
 }
 
+func (cm *controllerManager) GetRESTClientFor(obj runtime.Object) (rest.Interface, error) {
```

Comment:
Wouldn't this make more sense of the client? Ultimately, the use case for this is subresources that require a typed client, right? And can we even instantiate a typed client from a `rest.Client`?